command/ca/provisioner/caConfigClient.go

```diff
@@ -0,0 +1,202 @@
+package provisioner
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/smallstep/certificates/authority"
+	"github.com/smallstep/certificates/authority/config"
+	"github.com/smallstep/certificates/authority/provisioner"
+	"github.com/smallstep/certificates/ca"
+	"go.step.sm/cli-utils/ui"
+	"go.step.sm/linkedca"
+)
+
+// nodb implements the certificates/Admiclient interface with noops.
+type nodb struct{}
+
+func newNoDB() *nodb {
+	return &nodb{}
+}
+
+func (n *nodb) CreateProvisioner(ctx context.Context, prov *linkedca.Provisioner) error {
+	return nil
+}
+
+func (n *nodb) GetProvisioner(ctx context.Context, id string) (*linkedca.Provisioner, error) {
+	return nil, nil
+}
+
+func (n *nodb) GetProvisioners(ctx context.Context) ([]*linkedca.Provisioner, error) {
+	return nil, nil
+}
+
+func (n *nodb) UpdateProvisioner(ctx context.Context, prov *linkedca.Provisioner) error {
+	return nil
+}
+
+func (n *nodb) DeleteProvisioner(ctx context.Context, id string) error {
+	return nil
+}
+
+func (n *nodb) CreateAdmin(ctx context.Context, admin *linkedca.Admin) error {
+	return nil
+}
+
+func (n *nodb) GetAdmin(ctx context.Context, id string) (*linkedca.Admin, error) {
+	return nil, nil
+}
+
+func (n *nodb) GetAdmins(ctx context.Context) ([]*linkedca.Admin, error) {
+	return nil, nil
+}
+
+func (n *nodb) UpdateAdmin(ctx context.Context, prov *linkedca.Admin) error {
+	return nil
+}
+
+func (n *nodb) DeleteAdmin(ctx context.Context, id string) error {
+	return nil
+}
+
+type caConfigClient struct {
+	configFile string
+	ctx        context.Context
+	auth       *authority.Authority
+}
+
+func newCaConfigClient(ctx context.Context, cfg *config.Config, cfgFile string) (*caConfigClient, error) {
+	a, err := authority.New(cfg, authority.WithAdminDB(newNoDB()), authority.WithSkipInit())
```

Comment:
Just an idea, instead of this client, would it make sense to create an admin db that edits the ca.json?

Reply:
It's an interesting idea, but then you would have to http connect to an instance of `step-ca` to make changes to the configuration, if I'm understanding you correctly. Some of our users make modifications to the `ca.json` offline. We would break that functionality.

Reply:
I didn't mean that, I meant that having the implementation on the `step-ca` code base might make more sense.